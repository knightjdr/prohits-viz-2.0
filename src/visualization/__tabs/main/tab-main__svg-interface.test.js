import { shallow } from 'enzyme';

import renderInterface from './tab-main__svg-interface';

jest.mock('./tab-main__store-connection');
jest.mock('./tab-main__svg');

describe('SVG interface', () => {
  describe('when visible', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        renderInterface({
          other: 'otherProp',
          plotTranslate: 100,
          setContainerRef: { ref: 'test' },
          show: true,
        }),
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should set transform', () => {
      expect(wrapper.find('.heatmap-svg__wrapper').props().style.transform).toBe('translate(100px)');
    });

    it('should render store connection', () => {
      expect(wrapper.find('StoreConnection').length).toBe(1);
    });

    it('should pass other props to store connection', () => {
      const storeConnection = wrapper.find('StoreConnection');
      expect(Object.prototype.hasOwnProperty.call(storeConnection.props(), 'other')).toBeTruthy();
    });
  });

  describe('when hidden', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        renderInterface({
          other: 'otherProp',
          plotTranslate: 100,
          setContainerRef: { ref: 'test' },
          show: false,
        }),
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should show svg element', () => {
      expect(wrapper.find('StoreConnection').length).toBe(0);
    });
  });
});
