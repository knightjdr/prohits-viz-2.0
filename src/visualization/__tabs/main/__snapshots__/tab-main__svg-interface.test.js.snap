// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG interface when hidden should match snapshot 1`] = `
<div
  className="heatmap-svg__wrapper"
  style={
    Object {
      "transform": "translate(100px)",
    }
  }
/>
`;

exports[`SVG interface when visible should match snapshot 1`] = `
<div
  className="heatmap-svg__wrapper"
  style={
    Object {
      "transform": "translate(100px)",
    }
  }
>
  <StoreConnection
    other="otherProp"
    renderProp={[Function]}
  />
</div>
`;
